include/onnxruntime/core/graph/graph.h

```diff
@@ -974,6 +974,8 @@ class Graph {
   // NodeArgs that come from outer scope. Used when building a graph so that
   // these don't get recorded as graph inputs in the GraphProto.
   std::unordered_set<std::string> outer_scope_node_arg_names_;
+
+  int num_resolves_ = 0;
```

Comment:
Quick comment here that this tracks the number of times this graph has been resolved?
Also, what happens to this int when the graph gets (de)serialized? I guess it just gets reset since it is not part of the proto, right? Just wondering if this should be getting reset by the session state initializer, although I dont think so.

Reply:
No change when a graph is serialized. Set to zero when a graph is deserialized as we create a new instance of Graph at that point. It's an internal for the Graph class so session state initializer should have nothing to do with it.

Reply:
Yep, I agree.